File: backend/workspaces/sessions/default-candidate/typescript-task-manager/manager.ts
interface Task {
    id: number;
    title: string;
    priority: "low" | "medium" | "high";
    done: boolean;
}

class TaskManagerImpl {
    private tasks: Task[] = [];
    private nextId: number = 1;

    addTask(title: string, priority: "low" | "medium" | "high"): Task {
        // TODO: Create a new Task, push to tasks, increment nextId
        // Return the created task
        return { id: 0, title: "", priority: "low", done: false };
    }

    completeTask(taskId: number): void {
        // TODO: Find task by id and set done = true
    }

    removeTask(taskId: number): void {
        // TODO: Remove task by id from tasks
    }

    listTasks(): Task[] {
        // TODO: Return all tasks
        return [];
    }

    filterByPriority(priority: string): Task[] {
        // TODO: Return tasks matching the given priority
        return [];
    }
}

function createTaskManager() {
    return new TaskManagerImpl();
}

File: backend/workspaces/sessions/demo-session-5/manager.py
class TaskManager:
    """Manages a collection of tasks."""
    def __init__(self):
        self.tasks = []
        self.next_id = 1

    def add_task(self, title: str, priority: str) -> dict:
        # TODO: Create a new Task, append to self.tasks, increment next_id
        # Return the task as a dict using to_dict()
        pass

    def complete_task(self, task_id: int):
        # TODO: Find task by id and set done = True
        pass

    def remove_task(self, task_id: int):
        # TODO: Remove task by id from self.tasks
        pass

    def list_tasks(self) -> list:
        # TODO: Return all tasks as list of dicts
        pass

    def filter_by_priority(self, priority: str) -> list:
        # TODO: Return tasks matching the given priority as list of dicts
        pass
